demo/mbtaupsert/src/kafka_interaction.rs

```diff
@@ -0,0 +1,297 @@
+        }
+        Ok(())
+    }
+
+    pub fn create_consistency_topic(
+        &mut self,
+        topic_name: &str,
+        replication: i32,
+    ) -> Result<(), String> {
+        //disable deleting old segments of the stream
+        self.create_topic(
+            topic_name,
+            1,
+            replication,
+            vec![
+                ("cleanup.policy".to_owned(), "delete".to_owned()),
+                ("retention.ms".to_owned(), "-1".to_owned()),
+            ],
+        )
+    }
+
+    pub fn ingest(
+        &mut self,
+        topic_name: &str,
+        partition: i32,
+        key: Option<String>,
+        value: Option<String>,
+    ) -> Result<(), String> {
+        let val_buf = if let Some(value) = value {
+            value.as_bytes().to_vec()
+        } else {
+            Vec::new()
+        };
+        let key_buf = if let Some(key) = key {
+            key.as_bytes().to_vec()
+        } else {
+            Vec::new()
+        };
+        self.ingest_inner(topic_name, partition, key_buf, val_buf)
+    }
+
+    fn ingest_inner(
+        &mut self,
+        topic_name: &str,
+        partition: i32,
+        key_buf: Vec<u8>,
+        val_buf: Vec<u8>,
+    ) -> Result<(), String> {
+        let futs = FuturesUnordered::new();
+        let record: FutureRecord<_, _> = FutureRecord::to(&topic_name)
+            .payload(&val_buf)
+            .key(&key_buf)
+            .partition(partition);
+
+        futs.push(self.kafka_producer.send(record, Duration::from_secs(1)));
+        block_on(
+            futs.map_err(|(e, _)| e.to_string())
+                .try_for_each(|_| future::ok(())),
+        )
+    }
+
+    pub fn ingest_consistency(
+        &mut self,
+        consistency_name: &str,
+        topic_name: &str,
+        timestamp: i64,
+        offset: i64,
+    ) -> Result<(), String> {
+        let schema = &self.consistency_schema.as_ref().unwrap();
+        let mut val = Record::new(schema.top_node()).unwrap();
+        val.put("source", topic_name);
+        val.put("partition_count", 1i32);
+        val.put(
+            "partition_id",
+            Value::Union {
+                index: 0,
+                inner: Box::new(Value::Int(0i32)),
+                n_variants: 2,
+                null_variant: None,
+            },
+        );
+        val.put("timestamp", timestamp);
+        val.put("offset", offset);
+        let mut val_buf = vec![];
+        // The first byte is a magic byte (0) that indicates the Confluent
+        // serialization format version, and the next four bytes are a
+        // 32-bit schema ID.
+        //
+        // https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html#wire-format
+        val_buf.write_u8(0).unwrap();
+        val_buf.write_i32::<NetworkEndian>(1).unwrap();
+        val_buf.extend(avro::to_avro_datum(schema, val).map_err(|e| e.to_string())?);
+
+        self.ingest_inner(consistency_name, 0, vec![], val_buf)
+    }
+}
```

Comment:
This is way too much code to duplicate. We're trying to factor these methods into the `kafka-util` crate. The create topics function should already do what you need. We should figure out something for the message production too, because hacking up schema IDs here strikes me as really sketchy, and duplicates a lot of logic that already exists in `interchange`.

Reply:
I just commented out BYO support wholesale since it is not necessary for a user to try out.